shared/security/src/main/java/io/pravega/shared/security/token/JsonWebToken.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.security.token;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.SignatureException;
+import io.pravega.auth.InvalidTokenException;
+import io.pravega.auth.TokenException;
+import io.pravega.auth.TokenExpiredException;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import lombok.NonNull;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Creates Json Web Tokens (JWT). Also, parses a JWT.
+ */
+@ToString
+@Slf4j
+public class JsonWebToken {
```

Comment:
I'm not sure why you had to create this class rather than keep using `io.jsonwebtoken.Jwts`. Is my understanding right that we were using it before and that's the reason for removing all `Jwts` builder references?

Reply:
`io.pravega.shared.security.token.JsonWebToken` is a new class created in this PR for creating and parsing a JSON Web Token (JWT). This class internally uses `io.jsonwebtoken.Jwts` and encapsulates Pravega-specific logic for processing tokens. 

To explain the rationale for making this change a bit more, previously, the logic was spread and even duplicated in multiple places. It was hard to get a full picture of how Pravega creates and parses token and was more difficult than necessary to make any changes. Now, all JWT token-related domain logic is in one class. The change also made unit testing easier, and a bug was found in existing code and fixed. 

All other code in Pravega now uses this class for processing tokens, and therefore does not need any references to `io.jsonwebtoken.Jwts`.